MaxText/decode_checkpoint.py

```diff
@@ -0,0 +1,123 @@
+"""
```

Comment:
Did you discuss with Colin Gaffney if we could instead load just the params from a bigger checkpoint? I would have thought (possibly naively) that we could. (Also maybe this file would still be useful?)

Reply:
Totally reasonable! But wouldn't we still want to execute a partial read?